### DISCUSSION TITLE
I2C instances

### DISCUSSION THREAD

**Discussion by kowalski100** [polarfire-soc/polarfire-soc-documentation#55] [Bare metal embedded software] [answered]
The Peripheral Guide says Icicle has two instances of I2C buses. But in driver [here](https://github.com/polarfire-soc/polarfire-soc-bare-metal-library/blob/6a093aa9570180170a3874ecbed78a658666cf7d/src/platform/drivers/mss/mss_i2c/mss_i2c.c#L96), four instances are mentioned.

What exactly the low and high mean in the instance? i.e.
g_mss_i2c0_lo
g_mss_i2c0_hi

**Comment by Cyril-Jean** [comment, marked as answer]
Some of the "low speed" peripherals such as I2C can appear in one of two possible address ranges in the memory map, hi or lo. This is intended to make efficient use of the PMP registers to separate two AMP contexts.

**Comment by kowalski100** [comment]
Thanks Cyril-Jean.
